### PR TITLE
RavenDB-4151: Docker Support

### PR DESCRIPTION
**Pull request by jonstelly**
This includes some basic Dockerfiles for Docker, and some powershell scripts to build, run and deploy them.

There are probably a few things to change before accepting this pull request

- The docker images are currently built by downloading the zip files from the website.  It would probably make more sense to use artifacts generated during build (I could use some guidance on how best to do that)
- For Windows/NanoServer, vcruntime140.dll is now needed for sodium.  The vcredist executable won't run on NanoServer, so we can't just download the executable and run in the Dockerfile, but not sure if GitHub is the right place to keep that file.
- AllowEverybodyToAccessTheServerAsAdmin - I have this on for now so I can administer the raven instance, is there a better way to set that up?
- I imagine the scripts I'm using should be folded into the scripts directory, but I figured it was easier to review this pull request with all files in a single directory, and also those scripts are pretty well organized and I was sure I would put something in the wrong place
- Someone will need to register a docker organization, (ravendb?)  I didn't do that because I didn't want to misrepresent the builds I have uploaded [here](https://hub.docker.com/r/kymeric/ravendb/) as being official.

If someone can give me guidance on any of the above, I'm happy to clean this up to get this pull request accepted.

### CODE REVIEW

**Comment by ayende** [comment]
@jonstelly Can you send the CLA?

**Comment by jonstelly** [comment]
Hadn't clicked on the verify link.  Done now.

**Comment by ppekrol** [comment]
Thanks @jonstelly !
Tomorrow @gregolsky will be adjusting the scripts and incorporating them into our release process.

File: docker/ravendb-linux/Dockerfile [hunk 1]
```diff
@@ -0,0 +1,29 @@
+FROM ubuntu:xenial
+
+ENV AllowEverybodyToAccessTheServerAsAdmin true
```
Comment (talweiss1982) [inline docker/ravendb-linux/Dockerfile:3]: this must be changed

File: docker/ravendb-nanoserver/Dockerfile [hunk 2]
```diff
@@ -0,0 +1,13 @@
+FROM microsoft/dotnet:runtime-nanoserver
+
+ENV AllowEverybodyToAccessTheServerAsAdmin true
```
Comment (talweiss1982) [inline docker/ravendb-nanoserver/Dockerfile:3]: Also this

File: docker/ravendb-linux/Dockerfile (hunk 1 above)
Comment (gregolsky) [reply docker/ravendb-linux/Dockerfile:3]: ok

File: docker/ravendb-nanoserver/Dockerfile (hunk 2 above)
Comment (gregolsky) [reply docker/ravendb-nanoserver/Dockerfile:3]: ok

**Pull request merged**